mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -306,12 +307,20 @@ class KMeans @Since("1.5.0") (
 
   @Since("2.0.0")
   override def fit(dataset: Dataset[_]): KMeansModel = {
+    val handlePersistence = dataset.rdd.getStorageLevel == StorageLevel.NONE
+    fit(dataset, handlePersistence)
+  }
+
+  @Since("2.2.0")
+  protected def fit(dataset: Dataset[_], handlePersistence: Boolean): KMeansModel = {
```

Comment:
I like the change in terms of eliminating the extra fit() method (which is not done anywhere else in spark.ml), but I think you'll need to keep the original test for storage level ```val handlePersistence = dataset.rdd.getStorageLevel == StorageLevel.NONE```.  In the current change, the RDD being tested is different from ```dataset.rdd```, so handlePersistence would always be true regardless of the original storage level.  Could you please send a little PR which merges the 2 fit() methods?  Thanks!